custom_components/circadian_lighting/switch.py

```diff
@@ -243,120 +282,99 @@ def calc_hs(self):
+
+            service_data = {ATTR_ENTITY_ID: light}
+            brightness = self._brightness
+            if brightness is not None:
+                service_data[ATTR_BRIGHTNESS] = int((brightness / 100) * 254)
+            if transition is not None:
+                service_data[ATTR_TRANSITION] = transition
+
+            # Set color of array of ct.
+            if light in self._lights_ct:
+                which = "CT"
+                service_data[ATTR_COLOR_TEMP] = int(self.calc_ct())
+
+            # Set color of array of rgb.
+            elif light in self._lights_rgb:
+                which = "RGB"
+                service_data[ATTR_RGB_COLOR] = tuple(map(int, self.calc_rgb()))
+
+            # Set color of array of xy.
+            elif light in self._lights_xy:
+                which = "XY"
+                service_data[ATTR_XY_COLOR] = self.calc_xy()
+                if service_data.get(ATTR_BRIGHTNESS, False):
+                    service_data[ATTR_WHITE_VALUE] = service_data[ATTR_BRIGHTNESS]
+
+            # Set color of array of brightness.
+            elif light in self._lights_brightness:
+                which = "Brightness"
+
+            self.hass.services.call(LIGHT_DOMAIN, SERVICE_TURN_ON, service_data)
+            key_value_strings = [
+                f"{k}: {v}" for k, v in service_data.items() if k != ATTR_ENTITY_ID
+            ]
+            msg = ", ".join(key_value_strings)
+            _LOGGER.debug(f"{light} {which} Adjusted - {msg}")
 
     def light_state_changed(self, entity_id, from_state, to_state):
-        try:
-            _LOGGER.debug(entity_id + " change from " + str(from_state) + " to " + str(to_state))
-            if to_state.state == 'on' and from_state.state != 'on':
+        with suppress(Exception):
+            _LOGGER.debug(f"{entity_id} change from {from_state} to {to_state}")
+            if to_state.state == "on" and from_state.state != "on":
                 self.adjust_lights([entity_id], self._initial_transition)
-        except:
-            pass
 
     def sleep_state_changed(self, entity_id, from_state, to_state):
-        try:
-            _LOGGER.debug(entity_id + " change from " + str(from_state) + " to " + str(to_state))
-            if to_state.state == self._sleep_state or from_state.state == self._sleep_state:
-                self.update_switch(self._initial_transition)
-        except:
-            pass
-    
+        with suppress(Exception):
+            _LOGGER.debug(f"{entity_id} change from {from_state} to {to_state}")
+            if (
+                to_state.state == self._sleep_state
+                or from_state.state == self._sleep_state
+            ):
+                self._update_switch(self._initial_transition, force=True)
+
     def disable_state_changed(self, entity_id, from_state, to_state):
-        try:
-            _LOGGER.debug(entity_id + " change from " + str(from_state) + " to " + str(to_state))
+        with suppress(Exception):
```

Comment:
@claytonjn, do you know why you catch these errors?

Should they really occur, maybe if they do, it is a bug?

Reply:
I think they shouldn't occur, so I removed the catch all and I haven't seen the error yet.

Reply:
I _think_ the str() would throw an error on first state_change (after reboot) where from_state didn't exist, and this was my lazy way to handle that. The logging of those state-changes has been useful though in cases when users don't understand why lights are being adjusted - I'd like to add that back if possible.

Reply:
I have added a function to format the log better:
```
2020-09-04 16:22:32 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because light.bedroom_lights changed from off to on and these attributes changes: brightness: 39, hs_color: (254.685, 56.078), rgb_color: (147, 112, 255), xy_color: (0.235, 0.148).
2020-09-04 16:22:32 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because light.bedroom_lights changed from off to on and these attributes changes: brightness: 39, hs_color: (254.685, 56.078), rgb_color: (147, 112, 255), xy_color: (0.235, 0.148).
2020-09-04 16:22:43 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from off to half and no attributes changed.
2020-09-04 16:22:43 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from off to half and no attributes changed.
2020-09-04 16:22:43 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from off to half and no attributes changed.
2020-09-04 16:22:46 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from half to total and no attributes changed.
2020-09-04 16:22:46 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from half to total and no attributes changed.
2020-09-04 16:22:46 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from half to total and no attributes changed.
2020-09-04 16:22:47 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from total to off and no attributes changed.
2020-09-04 16:22:47 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from total to off and no attributes changed.
2020-09-04 16:22:47 DEBUG (MainThread) [custom_components.circadian_lighting.switch] Lights adjusting because input_select.sleep_mode changed from total to off and no attributes changed.
```

See 0264aeaa09af34c05c4e1a918330bb91344c5f82.